docs/architecture/adr-019-subspace-post-moving.md

```diff
@@ -0,0 +1,77 @@
+# ADR 019: Subspace post migration
+
+## Changelog
+- April 20th, 2023: First draft;
+
+## Status
+
+Proposed
+
+## Abstract
+
+This ADR introduces a new feature that enables post owners to move their own content between subspaces.
+
+## Context
+
+Currently, the only way to move a post to another subspace is by following process:
+1. Create an identical post as the original post in the target subspace
+2. Delete the original post on source subspace
+
+This process is overly complicated and time-consuming because it requires the user to perform two steps and manually recreate the post in the new location. Additionally, it can lead to inconsistencies if the user forgets to include certain details. A more user-friendly process would be beneficial for improving efficiency and reducing the risk of inconsistencies.
+
+## Decision
+
+We will implement a new handler that allows users to move their own posts through a message. Here are the upcoming changes:
+
+### `Msg` Service
+
+```proto
+service Msg {
+    // MovePost allows users to move their own posts to another subspace
+    rpc MovePost(MsgMovePost) returns (MsgMovePostResponse);
```

Comment:
Should we use a new `MsgMovePost` message, or should we instead use the existing `MsgEditPost`? I think there are pro/cons for each one but I'm not sure if there's a clear winner on this matter. What do you think?

Reply:
I think defining a new message can have the benefit as follows:
1. User can avoid from migrating post to an another subspace accidentally when editing a post, and vice versa
2. `MsgEditPost` related methods inside desmos-bindings, desmjs can stay away from breaking change
3. Tracker can easily track the migration history of a post

As result, I decided to define a new message.